### PR TITLE
pkg/steps: add build pending timeout period

### PR DESCRIPTION
**Pull request by bbguimaraes**
https://github.com/openshift/ci-tools/pull/2875
https://issues.redhat.com/browse/DPTP-2836

Based on https://github.com/openshift/ci-tools/pull/3385.

/hold

### CODE REVIEW

File: pkg/kubernetes/client.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func WaitForConditionOnObject(ctx context.Context, client ctrlruntimeclient.With
 
 type PodClient interface {
 	loggingclient.LoggingClient
-	PendingTimeout() time.Duration
+	GetPendingTimeout() time.Duration
```
Comment (danilo-gemoli) [inline pkg/kubernetes/client.go:74]: If I've understood it correctly this gets the value from `--pod-pending-timeout` and it controls a pod lifecycle regardless its nature: being it a test pod, a build, etc. Is it enough? Do we need it to be more fine-grained (I mean, having more than one value)?

**Comment by danilo-gemoli** [comment]
Other than the scary commits history, which could be now simplified as #3385 has been merged, it looks fine overall.

File: pkg/kubernetes/client.go (hunk 1 above)
Comment (bbguimaraes) [reply pkg/kubernetes/client.go:74]: Yes, that is correct, this is the same period as is used for pods in general.

The causes for scheduling delays are for the most part the same for all types of pods, so I don't see the need to add a separate value for builds.

**Comment by bbguimaraes** [comment]
Rebased.

/hold cancel

**Comment by bbguimaraes** [comment]
> ```
> INFO[2023-04-24T15:04:47Z] pod pending for more than 30s: container "ci-scheduling-dns-wait" has not started in 1m58.8714557s: 
> …
> ```

It may also be good to change the error message to include all pending containers, it's not immediately obvious that the scheduling requirements of one of the (non-init.) containers is the cause of the problem.

**Pull request merged**